actionpack/lib/action_dispatch/middleware/server_timing.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+require "action_dispatch/middleware/server_timing_subscriber"
+
+module ActionDispatch
+  class ServerTiming
+    SERVER_TIMING_HEADER = "Server-Timing"
+
+    class << self
+      def add_timing(key:, dur: nil, desc: nil)
+        timings[key] = { dur: dur, desc: desc }
+      end
+
+      def to_header
+        metrics = []
+        timings.each_pair do |key, data|
+          metric = +key.to_s
+          metric << ";desc=\"#{data[:desc]}\"" if data[:desc]
+          metric << ";dur=#{data[:dur]}" if data[:dur]
+          metrics << metric
+        end
+        metrics.join(", ")
+      end
+
+      def measure(key, desc: nil, &blk)
```

Comment:
This is only a convenience method, the other way of specifying this is to do something like:

``` ruby
ActiveSupport::Notifications.instrument("custom_event", server_timing: { key: "my_key", des: "select_all" }) do
 ...
end
```

As oposed to: 
``` ruby
ServerTiming.measure("my_key", desc: "select_all") do
..
end
```

We can chose to remove this method, if we don't find it useful. If we keep it, I'd rather move away from using `ActiveSupport::Notifications.instrument` within it, and rather implement it as: 

``` ruby
def measure(key, desc: nil, &blk)
  duration = Benchmark.ms { yield }
  ServerTiming.add_timing(key: key, desc: desc, dur: duration)
end
```

This lets us get rid of the `ServerTimingSubscriber` and is a much simpler implementation.